### PR TITLE
Notification and Issue on Vulnerability Scan Failure

### PR DESCRIPTION
**Pull request by zhijie-yang**
- [x] I have signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] I am following the [CONTRIBUTING](https://github.com/canonical/oci-factory/blob/main/CONTRIBUTING.md) guidelines

*Ping the @canonical/rocks team.*

---

## Description

This PR gives the OCI Factory the ability to create issues in the repository of the built rock, if there are failing vulnerability scanning results, keep them updated with the latest scanning results (if changes exist), and notify in the MM channel if new vulnerabilities are found.

### Related issues
ROCKS-1330

## Off-The-Fly tests
1. Should create the issue if CVE is found, and the issue does not exist: [run](https://github.com/zhijie-yang/workflow-playground/actions/runs/10613017167/job/29415951425#step:7:26)
2. Should skip (nop) if no new CVE found, while the issue exists: [run](https://github.com/zhijie-yang/workflow-playground/actions/runs/10613017167/job/29416004996#step:7:22)
3. Should update the issue if new CVE is published (should also notify in MM, no changes comparing to the original code for notifying in MM, not tested off-the-fly): [run](https://github.com/zhijie-yang/workflow-playground/actions/runs/10613068793/job/29416120264#step:7:28)
4. Should close the issue if no CVE is found while the issue exists: [run](https://github.com/zhijie-yang/workflow-playground/actions/runs/10613081637/job/29416156362#step:8:10)
5. Should skip steps `Create issue` and `Close issue` if no CVE is found, and the issue does not exist: [run](https://github.com/zhijie-yang/workflow-playground/actions/runs/10613081637/job/29416239874)
6. Raise issues in other repo: [run](https://github.com/zhijie-yang/workflow-playground/actions/runs/10614926324/job/29421904520#step:7:26), [issue](https://github.com/zhijie-yang/grafana-agent-rock/issues/1).

### CODE REVIEW

**Comment by zhijie-yang** [comment]
The corresponding `_Test OCI Factory | mock-rock` workflow run should be [this](https://github.com/canonical/oci-factory/actions/runs/10613059951).

File: .github/workflows/Vulnerability-Scan.yaml [hunk 1]
```diff
@@ -169,9 +176,90 @@ jobs:
...
+      # |      T       |   T    |          T           | update |
+      # |      T       |   T    |          F           |  never |
+      # |      T       |   F    |          T           |   nop  |
+      # |      T       |   F    |          F           |  close |
+      # |      F       |   T    |          T           | create |
+      # |      F       |   T    |          F           |  never |
+      # |      F       |   F    |          T           | create |
+      # |      F       |   F    |          F           |   nop  |
+      
+      - name: Create issue
```
Comment (cjdcordeiro) [inline .github/workflows/Vulnerability-Scan.yaml:245]:
```suggestion
      - name: Notify via GitHub issue
```

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]

File: .github/workflows/Vulnerability-Scan.yaml [hunk 2]
```diff
@@ -169,9 +176,90 @@ jobs:
...
+          echo '${{ needs.test-vulnerabilities.outputs.vulnerabilities }}' | jq -r '.[] | "| \(.VulnerabilityID) | /\(.Target) | \(.Severity) | \(.PkgName) |"' >> issue.md
+          echo -e "\nDetails: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}" >> issue.md
+          num_vulns=$(echo '${{ needs.test-vulnerabilities.outputs.vulnerabilities }}' | jq -r 'length')
+          echo "issue-title=$title" >> "$GITHUB_OUTPUT"
+          echo "issue-body-file=issue.md" >> "$GITHUB_OUTPUT"
+          echo "vulnerability-exists=$([[ $num_vulns -gt 0 ]] && echo 'true' || echo 'false')" >> "$GITHUB_OUTPUT"
+
+      - id: issue-exists
+        run: |
+          issue_number=$(gh issue list --repo github.com/zhijie-yang/workflow-playground --json "number,title" \
```
[the remark this answers is not in the document]
Comment (zhijie-yang) [reply .github/workflows/Vulnerability-Scan.yaml]: Fixed. Thanks for pointing out.

**Review by cjdcordeiro — APPROVED** [review]
Thanks. LGTM
waiting on @clay-lake

File: .github/workflows/Vulnerability-Scan.yaml [hunk 3]
```diff
@@ -169,9 +176,90 @@ jobs:
...
+          echo "img-repo=$img_repo" >> "$GITHUB_OUTPUT"
+
+      # We have to walk through the vulnerabilities since trivy does not support outputting the results as Markdown
+      - name: Create Markdown Content
+        id: create-markdown
+        run: |
+          set -x
+          title="Vulnerabilities found for ${{ steps.simplify-image-name.outputs.img_name }}"
+          echo "## $title" > issue.md
+          echo "| ID | Target |Severity | Package |" >> issue.md
```
Comment (zhijie-yang) [inline .github/workflows/Vulnerability-Scan.yaml:216]:
```suggestion
          echo "| ID | Target | Severity | Package |" >> issue.md
```

**Review by clay-lake — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**